roman_imsim/skycat.py

```diff
@@ -225,14 +221,22 @@ def getObj(self, index, gsparams=None, rng=None, exptime=30):
         else:
             gs_object = galsim.Add(gs_obj_list)
 
+        # This should catch both "star" and "gaia_star" objects
+        if "star" in skycat_obj.object_type:
+            # Cap (star) flux at 30M photons to avoid gross artifacts when trying
+            # to draw the Roman PSF in finite time and memory
+            flux_cap = 3e7
+            if flux > flux_cap:
+                flux = flux_cap
+
         # Give the object the right flux
+        gs_object = gs_object.withFlux(flux, self.bandpass)
```

Comment:
Is it necessary to set the `flux` attribute below after you've set it with `withFlux` here?